library/core/src/mem/mod.rs

```diff
@@ -1083,14 +1083,22 @@ impl<T> clone::Clone for Discriminant<T> {
 }
 
 #[stable(feature = "discriminant_value", since = "1.21.0")]
-impl<T> cmp::PartialEq for Discriminant<T> {
+#[rustc_const_unstable(feature = "const_partial_eq_discriminant", issue = "none")]
+impl<T> const cmp::PartialEq for Discriminant<T>
+where
+    <T as DiscriminantKind>::Discriminant: ~const cmp::PartialEq,
```

Comment:
Not entirely sure if this bounds should be here or if we should add the bound to `DiscriminantKind::Discriminant`, but that would require the Implementations of `DiscriminantKind` to be marked as const.